packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m

```diff
@@ -0,0 +1,244 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+#import <EXScreenOrientation/EXScreenOrientationModule.h>
+#import <UMCore/UMEventEmitterService.h>
+#import <EXScreenOrientation/EXScreenOrientationRegistry.h>
+#import <UMCore/UMModuleRegistryProvider.h>
+#import <EXScreenOrientation/EXScreenOrientationUtilities.h>
+
+#import <UIKit/UIKit.h>
+
+@interface EXScreenOrientationModule ()
+
+@property (nonatomic, weak) UMModuleRegistry *moduleRegistry;
+@property (nonatomic, weak) id<UMEventEmitterService> eventEmitter;
+@property (nonatomic) bool hasListeners;
+@property (nonatomic) UIInterfaceOrientation currentScreenOrientation;
+
+@end
+
+static int INVALID_MASK = 0;
+static NSString *defaultAppId = @"bareAppId";
+
+@implementation EXScreenOrientationModule
+
+UM_EXPORT_MODULE(ExpoScreenOrientation);
+
+- (instancetype)init
+{
+  if ((self = [super init])) {
+    [[UIDevice currentDevice] beginGeneratingDeviceOrientationNotifications];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(handleDeviceOrientationChange:)
+                                                 name:UIDeviceOrientationDidChangeNotification
+                                               object:[UIDevice currentDevice]];
+  }
+  return self;
+}
+
+- (void)dealloc
+{
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+  _moduleRegistry = moduleRegistry;
+  _eventEmitter = [moduleRegistry getModuleImplementingProtocol:@protocol(UMEventEmitterService)];
+  _hasListeners = NO;
+  UIDevice *device = [UIDevice currentDevice];
+  UIInterfaceOrientation currentDeviceOrientation = [EXScreenOrientationUtilities UIDeviceOrientationToUIInterfaceOrientation:[device orientation]];
+  UIInterfaceOrientationMask orientationMask = [self orientationMask];
+  
+  // this gives the correct information of screen orientation before any rotation or locking
+  if ([EXScreenOrientationUtilities doesOrientationMask:orientationMask containOrientation:currentDeviceOrientation]) {
+    _currentScreenOrientation = currentDeviceOrientation;
+  } else {
+    _currentScreenOrientation = [EXScreenOrientationUtilities defaultOrientationForOrientationMask:[self orientationMask]];
+  }
+}
+
+UM_EXPORT_METHOD_AS(lockAsync,
+                    lockAsync:(NSString *)orientationLockString
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  EXOrientationLock orientationLock = [EXScreenOrientationUtilities stringToOrientationLock:orientationLockString];
+  UIInterfaceOrientationMask orientationMask = [EXScreenOrientationUtilities orientationLockJSToNative:orientationLock];
+  if (orientationMask == INVALID_MASK) {
```

Comment:
I think I would be ok with just `if (!orientationMask)` which is more idiomatic and doesn't introduce another variable which (sic!) is defined in two separate files so can be changed too easily. Making explicit `return 0` and `if orientationMask > 0` makes `0` part of the API and not `INVALID_MASK`